provider/core/src/resource.rs

```diff
@@ -94,59 +113,27 @@ impl ResourceKey {
         self.hash
     }
 
-    /// Creates a new ResourceKey from a path, returning an error if the path is invalid.
-    ///
-    /// It is intended that `ResourceKey` objects are const-constructed. To force construction
-    /// into a const context, use [`resource_key!()`]. Doing so ensures that compile-time key
-    /// extraction functions as expected.
-    ///
-    /// # Example
-    ///
-    /// ```
-    /// use icu_provider::prelude::*;
-    /// use icu_provider::tagged;
-    ///
-    /// // Const constructed (preferred):
-    /// const k1: ResourceKey = icu_provider::resource_key!("foo/bar@1");
-    ///
-    /// // Runtime constructed:
-    /// let k2: ResourceKey = ResourceKey::try_new(tagged!("foo/bar@1")).unwrap();
-    ///
-    /// assert_eq!(k1, k2);
-    /// ```
-    #[inline]
-    pub const fn try_new(path: &'static str) -> Result<Self, DataError> {
-        match Self::check_path_syntax(path) {
-            Ok(_) => Ok(Self {
-                path,
-                hash: ResourceKeyHash::compute_from_str(path),
-            }),
-            Err(_) => Err(DataError::custom("resource key syntax error")),
-        }
-    }
-
-    const fn check_path_syntax(path: &str) -> Result<(), ()> {
-        let path = path.as_bytes();
-
-        // Start and end of the untagged part
+    #[doc(hidden)]
+    pub const fn try_new(path: &'static str) -> Option<Self> {
```

Comment:
nit: i would give this a name that is less easy to accidentally use (since it requires the tag), perhaps `construct_internal()`